x-pack/plugins/security/server/routes/api_keys/create.ts

```diff
@@ -19,18 +69,7 @@ export function defineCreateApiKeyRoutes({
     {
       path: '/internal/security/api_key',
       validate: {
-        body: schema.object({
-          name: schema.string(),
-          expiration: schema.maybe(schema.string()),
-          role_descriptors: schema.recordOf(
-            schema.string(),
-            schema.object({}, { unknowns: 'allow' }),
-            {
-              defaultValue: {},
-            }
-          ),
-          metadata: schema.maybe(schema.object({}, { unknowns: 'allow' })),
-        }),
+        body: schema.oneOf([bodySchema, bodySchemaWithKibanaPrivileges]),
```

Comment:
I was going to say we need to add unit test cases for the payload validation changes, but it turns out our unit tests are written so that they bypass validation 🙈 

Could you please add a TODO to the unit test file?

```diff
diff --git a/x-pack/plugins/security/server/routes/api_keys/create.test.ts b/x-pack/plugins/security/server/routes/api_keys/create.test.ts
index dbf4c93639e..70262d59bad 100644
--- a/x-pack/plugins/security/server/routes/api_keys/create.test.ts
+++ b/x-pack/plugins/security/server/routes/api_keys/create.test.ts
@@ -71,6 +71,7 @@ describe('Create API Key route', () => {
   });
 
   describe('success', () => {
+    test.todo('actually exercise different types of payload validation');
     test('allows an API Key to be created', async () => {
       authc.apiKeys.create.mockResolvedValue({
         api_key: 'abc123',
```